fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluid/v1/world/SidedFluidContainer.java

```diff
@@ -0,0 +1,66 @@
+package net.fabricmc.fabric.api.fluid.v1.world;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.FluidDrainable;
+import net.minecraft.block.FluidFillable;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.FluidState;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.BlockView;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldAccess;
+
+import net.fabricmc.fabric.api.fluid.v1.Action;
+import net.fabricmc.fabric.api.fluid.v1.container.FluidContainer;
+import net.fabricmc.fabric.api.fluid.v1.container.volume.FluidVolume;
+import net.fabricmc.fabric.api.fluid.v1.container.volume.SimpleFluidVolume;
+import net.fabricmc.fabric.api.fluid.v1.math.Drops;
+
+/**
+ * A container with different fluid containers per face, can be implemented on entities, and blocks (does not check block entities!).
+ */
+public interface SidedFluidContainer extends FluidFillable, FluidDrainable {
+	@Override
+	default Fluid tryDrainFluid(WorldAccess world, BlockPos pos, BlockState state) {
```

Comment:
would again prefer just passing World here, or is there a getWorld or asWorld in WorldAccess?

Reply:
mojang method